Developers/Unity Project/Spooksville/Assets/Scripts/Items/CandyCaneShooter.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CandyCaneShooter : Weapon
```

Comment:
If you need to code custom behavior for each weapon then having a class hierarchy might make sense. But if it's mostly just stats, assets etc. it may be overkill. In that case you could just make one `Weapon` class with a `WeaponType` enum or something that could be set to `CandyCaneShooter` or whatever, and have some stats for each instance. Put them in an array somewhere so you can easily edit them in the editor.

If there are going to be a lot of them it may also make sense to pull into an external data file such as a JSON or csv file. You can have a look at [the inventory system I wrote for Global Game Jam 2019](https://github.com/fadookie/ggj2019/tree/master/Assets/Scripts/Data) for an example of that.